controller-server/src/main/java/com/yahoo/vespa/hosted/controller/ApplicationController.java

```diff
@@ -332,33 +332,28 @@ else if ( ! application.deploying().isPresent() && ! zone.environment().isManual
                                                        " the current version " + existingDeployment.version());
             }
 
-            Optional<Rotation> rotation;
-            try (RotationLock rotationLock = rotationRepository.lock()) {
-                rotation = getRotation(application, zone, rotationLock);
-                if (rotation.isPresent()) {
-                    application = application.with(rotation.get().id());
+            Set<String> rotationNames = new HashSet<>();
+            Set<String> cnames = new HashSet<>();
+            if (zone.environment() == Environment.prod && application.deploymentSpec().globalServiceId().isPresent()) {
```

Comment:
`deployApplication` keeps growing. I would extract this entire block into a method, e.g. `private Application assignRotation(Application, Zone)` which returns the updated application.

Both Sets can be populated from `application.rotation()`.